fv3core/stencils/d_sw.py

```diff
@@ -603,44 +618,48 @@ def __init__(
         self.ytp_v = YTP_V(
             grid_indexing=grid_indexing,
             grid_data=grid_data,
-            grid_type=namelist.grid_type,
-            jord=namelist.hord_mt,
+            grid_type=grid_type,
+            jord=hord_mt,
         )
         self.xtp_u = XTP_U(
             grid_indexing=grid_indexing,
             grid_data=grid_data,
-            grid_type=namelist.grid_type,
-            iord=namelist.hord_mt,
+            grid_type=grid_type,
+            iord=hord_mt,
         )
         self.divergence_damping = DivergenceDamping(
             grid_indexing,
             grid_data,
             damping_coefficients,
-            self.grid.nested,
-            self.grid.stretched_grid,
-            spec.namelist.dddmp,
-            spec.namelist.d4_bg,
-            spec.namelist.nord,
-            spec.namelist.grid_type,
+            nested,
+            stretched_grid,
+            dddmp,
+            d4_bg,
+            nord,
+            grid_type,
             column_namelist["nord"],
             column_namelist["d2_divg"],
         )
-        full_origin = self.grid.full_origin()
-        full_domain = self.grid.domain_shape_full()
-        ax_offsets_full = axis_offsets(self.grid, full_origin, full_domain)
-        b_origin = self.grid.compute_origin()
-        b_domain = self.grid.domain_shape_compute(add=(1, 1, 0))
-        ax_offsets_b = axis_offsets(self.grid, b_origin, b_domain)
+        full_origin = grid_indexing.origin_full()
```

Comment:
Only because that's what the code I was refactoring did.

The whole pattern feels a little clunky for how often we use hit, honestly. I'm spending most of my time in these refactors trying to refactor and sometimes combine the many different ways we have to do this operation. What I mean is, 1) determine origin and domain in 1-2 function calls, 2) determine axis_offsets from origin and domain, 3) pass origin, domain, and axis_offsets into a stencil. Future plans, but I think we may want to combine all 3 into a single "initialize a stencil wrapper" call, along the lines of:

```python
# stencil_config is container class with GridIndexing, and gt4py configuration
stencil = stencil_config.Stencil(
    stencil_definition,
    dims=[X_DIM, Y_DIM, Z_DIM],
    halos=(2, 2),
    externals={"inline_q": inline_q},  # axis_offsets are automatically added
)
```

Alternatively, as a call to `fv3core.Stencil` where stencil_config is a required argument.

Reply:
sounds good.